
//---------------------------------------------------------------------
// This file is automatically generated. All manual edits will be lost.
//---------------------------------------------------------------------
// Proto file for the NI-FPGA Metadata
//---------------------------------------------------------------------
syntax = "proto3";

option cc_enable_arenas = true;
option java_multiple_files = true;
option java_package = "com.ni.grpc.nifpga";
option java_outer_classname = "NiFpga";
option csharp_namespace = "NationalInstruments.Grpc.NiFpga";

package nifpga_grpc;

import "session.proto";
import "data_moniker.proto";

service NiFpga {
  rpc Open(OpenRequest) returns (OpenResponse);
  rpc Close(CloseRequest) returns (CloseResponse);
  rpc Run(RunRequest) returns (RunResponse);
  rpc Reset(ResetRequest) returns (ResetResponse);
  rpc Download(DownloadRequest) returns (DownloadResponse);
  rpc ReadBool(ReadBoolRequest) returns (ReadBoolResponse);
  rpc BeginReadBool(BeginReadBoolRequest) returns (BeginReadBoolResponse);
  rpc ReadI8(ReadI8Request) returns (ReadI8Response);
  rpc BeginReadI8(BeginReadI8Request) returns (BeginReadI8Response);
  rpc ReadU8(ReadU8Request) returns (ReadU8Response);
  rpc BeginReadU8(BeginReadU8Request) returns (BeginReadU8Response);
  rpc ReadI16(ReadI16Request) returns (ReadI16Response);
  rpc BeginReadI16(BeginReadI16Request) returns (BeginReadI16Response);
  rpc ReadU16(ReadU16Request) returns (ReadU16Response);
  rpc BeginReadU16(BeginReadU16Request) returns (BeginReadU16Response);
  rpc ReadI32(ReadI32Request) returns (ReadI32Response);
  rpc BeginReadI32(BeginReadI32Request) returns (BeginReadI32Response);
  rpc ReadU32(ReadU32Request) returns (ReadU32Response);
  rpc BeginReadU32(BeginReadU32Request) returns (BeginReadU32Response);
  rpc ReadI64(ReadI64Request) returns (ReadI64Response);
  rpc BeginReadI64(BeginReadI64Request) returns (BeginReadI64Response);
  rpc ReadU64(ReadU64Request) returns (ReadU64Response);
  rpc BeginReadU64(BeginReadU64Request) returns (BeginReadU64Response);
  rpc WriteBool(WriteBoolRequest) returns (WriteBoolResponse);
  rpc BeginWriteBool(BeginWriteBoolRequest) returns (BeginWriteBoolResponse);
  rpc WriteI8(WriteI8Request) returns (WriteI8Response);
  rpc BeginWriteI8(BeginWriteI8Request) returns (BeginWriteI8Response);
  rpc WriteU8(WriteU8Request) returns (WriteU8Response);
  rpc BeginWriteU8(BeginWriteU8Request) returns (BeginWriteU8Response);
  rpc WriteI16(WriteI16Request) returns (WriteI16Response);
  rpc BeginWriteI16(BeginWriteI16Request) returns (BeginWriteI16Response);
  rpc WriteU16(WriteU16Request) returns (WriteU16Response);
  rpc BeginWriteU16(BeginWriteU16Request) returns (BeginWriteU16Response);
  rpc WriteI32(WriteI32Request) returns (WriteI32Response);
  rpc BeginWriteI32(BeginWriteI32Request) returns (BeginWriteI32Response);
  rpc WriteU32(WriteU32Request) returns (WriteU32Response);
  rpc BeginWriteU32(BeginWriteU32Request) returns (BeginWriteU32Response);
  rpc WriteI64(WriteI64Request) returns (WriteI64Response);
  rpc BeginWriteI64(BeginWriteI64Request) returns (BeginWriteI64Response);
  rpc WriteU64(WriteU64Request) returns (WriteU64Response);
  rpc BeginWriteU64(BeginWriteU64Request) returns (BeginWriteU64Response);
  rpc ReadArrayI16(ReadArrayI16Request) returns (ReadArrayI16Response);
  rpc BeginReadArrayI16(BeginReadArrayI16Request) returns (BeginReadArrayI16Response);
  rpc ReadArrayU16(ReadArrayU16Request) returns (ReadArrayU16Response);
  rpc BeginReadArrayU16(BeginReadArrayU16Request) returns (BeginReadArrayU16Response);
  rpc ReadArrayI64(ReadArrayI64Request) returns (ReadArrayI64Response);
  rpc BeginReadArrayI64(BeginReadArrayI64Request) returns (BeginReadArrayI64Response);
  rpc ReadArrayU64(ReadArrayU64Request) returns (ReadArrayU64Response);
  rpc BeginReadArrayU64(BeginReadArrayU64Request) returns (BeginReadArrayU64Response);
  rpc WriteArrayI16(WriteArrayI16Request) returns (WriteArrayI16Response);
  rpc BeginWriteArrayI16(BeginWriteArrayI16Request) returns (BeginWriteArrayI16Response);
  rpc WriteArrayU16(WriteArrayU16Request) returns (WriteArrayU16Response);
  rpc BeginWriteArrayU16(BeginWriteArrayU16Request) returns (BeginWriteArrayU16Response);
  rpc WriteArrayI64(WriteArrayI64Request) returns (WriteArrayI64Response);
  rpc BeginWriteArrayI64(BeginWriteArrayI64Request) returns (BeginWriteArrayI64Response);
  rpc WriteArrayU64(WriteArrayU64Request) returns (WriteArrayU64Response);
  rpc BeginWriteArrayU64(BeginWriteArrayU64Request) returns (BeginWriteArrayU64Response);
}

enum NiFpgaAttribute {
  NIFPGA_ATTRIBUTE_UNSPECIFIED = 0;
}

enum OpenAttribute {
  OPEN_ATTRIBUTE_UNSPECIFIED = 0;
  OPEN_ATTRIBUTE_NoRun = 1;
  OPEN_ATTRIBUTE_BitfilePathIsUTF8 = 2;
  OPEN_ATTRIBUTE_BitfileContentsNotPath = 3;
  OPEN_ATTRIBUTE_IgnoreSignatureArgument = 4;
}

enum CloseAttribute {
  CLOSE_ATTRIBUTE_UNSPECIFIED = 0;
  CLOSE_ATTRIBUTE_NoResetIfLastSession = 1;
}

enum RunAttribute {
  RUN_ATTRIBUTE_UNSPECIFIED = 0;
  RUN_ATTRIBUTE_WaitUntilDone = 1;
}

message I32Data {
  int32 value = 1;
}

message U32Data {
  uint32 value = 1;
}

message I64Data {
  int64 value = 1;
}

message U64Data {
  uint64 value = 1;
}

message BoolData {
  bool value = 1;
}

message ArrayI32Data {
  repeated int32 value = 1;
}

message ArrayU32Data {
  repeated uint32 value = 1;
}

message ArrayI64Data {
  repeated int64 value = 1;
}

message ArrayU64Data {
  repeated uint64 value = 1;
}

message OpenRequest {
  string session_name = 1;
  string bitfile = 2;
  string signature = 3;
  string resource = 4;
  oneof attribute_enum {
    OpenAttribute attribute_mapped = 5;
    uint32 attribute_raw = 6;
  }
}

message OpenResponse {
  int32 status = 1;
  nidevice_grpc.Session session = 2;
}

message CloseRequest {
  nidevice_grpc.Session session = 1;
  oneof attribute_enum {
    CloseAttribute attribute = 2;
    uint32 attribute_raw = 3;
  }
}

message CloseResponse {
  int32 status = 1;
}

message RunRequest {
  nidevice_grpc.Session session = 1;
  oneof attribute_enum {
    RunAttribute attribute = 2;
    uint32 attribute_raw = 3;
  }
}

message RunResponse {
  int32 status = 1;
}

message ResetRequest {
  nidevice_grpc.Session session = 1;
}

message ResetResponse {
  int32 status = 1;
}

message DownloadRequest {
  nidevice_grpc.Session session = 1;
}

message DownloadResponse {
  int32 status = 1;
}

message ReadBoolRequest {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadBoolResponse {
  int32 status = 1;
  bool value = 2;
}

message BeginReadBoolRequest {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadBoolResponse {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadI8Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadI8Response {
  int32 status = 1;
  int32 value = 2;
}

message BeginReadI8Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadI8Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadU8Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadU8Response {
  int32 status = 1;
  uint32 value = 2;
}

message BeginReadU8Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadU8Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadI16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadI16Response {
  int32 status = 1;
  int32 value = 2;
}

message BeginReadI16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadI16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadU16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadU16Response {
  int32 status = 1;
  uint32 value = 2;
}

message BeginReadU16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadU16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadI32Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadI32Response {
  int32 status = 1;
  int32 value = 2;
}

message BeginReadI32Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadI32Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadU32Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadU32Response {
  int32 status = 1;
  uint32 value = 2;
}

message BeginReadU32Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadU32Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadI64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadI64Response {
  int32 status = 1;
  int64 value = 2;
}

message BeginReadI64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadI64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadU64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message ReadU64Response {
  int32 status = 1;
  uint64 value = 2;
}

message BeginReadU64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
}

message BeginReadU64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteBoolRequest {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  bool value = 3;
}

message WriteBoolResponse {
  int32 status = 1;
}

message BeginWriteBoolRequest {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
}

message BeginWriteBoolResponse {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteI8Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  int32 value = 3;
}

message WriteI8Response {
  int32 status = 1;
}

message BeginWriteI8Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
}

message BeginWriteI8Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteU8Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 value = 3;
}

message WriteU8Response {
  int32 status = 1;
}

message BeginWriteU8Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
}

message BeginWriteU8Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteI16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  int32 value = 3;
}

message WriteI16Response {
  int32 status = 1;
}

message BeginWriteI16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
}

message BeginWriteI16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteU16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 value = 3;
}

message WriteU16Response {
  int32 status = 1;
}

message BeginWriteU16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
}

message BeginWriteU16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteI32Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  int32 value = 3;
}

message WriteI32Response {
  int32 status = 1;
}

message BeginWriteI32Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  int32 value = 3;
}

message BeginWriteI32Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteU32Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 value = 3;
}

message WriteU32Response {
  int32 status = 1;
}

message BeginWriteU32Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
}

message BeginWriteU32Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteI64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  int64 value = 3;
}

message WriteI64Response {
  int32 status = 1;
}

message BeginWriteI64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  int64 value = 3;
}

message BeginWriteI64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteU64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint64 value = 3;
}

message WriteU64Response {
  int32 status = 1;
}

message BeginWriteU64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint64 value = 3;
}

message BeginWriteU64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadArrayI16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message ReadArrayI16Response {
  int32 status = 1;
  repeated int32 array = 2;
}

message BeginReadArrayI16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message BeginReadArrayI16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadArrayU16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message ReadArrayU16Response {
  int32 status = 1;
  repeated uint32 array = 2;
}

message BeginReadArrayU16Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message BeginReadArrayU16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadArrayI64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message ReadArrayI64Response {
  int32 status = 1;
  repeated int64 array = 2;
}

message BeginReadArrayI64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message BeginReadArrayI64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message ReadArrayU64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message ReadArrayU64Response {
  int32 status = 1;
  repeated uint64 array = 2;
}

message BeginReadArrayU64Request {
  nidevice_grpc.Session session = 1;
  uint32 indicator = 2;
  uint32 size = 3;
}

message BeginReadArrayU64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteArrayI16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  repeated int32 array = 3;
  uint32 size = 4;
}

message WriteArrayI16Response {
  int32 status = 1;
}

message BeginWriteArrayI16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 size = 3;
}

message BeginWriteArrayI16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteArrayU16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  repeated uint32 array = 3;
  uint32 size = 4;
}

message WriteArrayU16Response {
  int32 status = 1;
}

message BeginWriteArrayU16Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 size = 3;
}

message BeginWriteArrayU16Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteArrayI64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  repeated int64 array = 3;
  uint32 size = 4;
}

message WriteArrayI64Response {
  int32 status = 1;
}

message BeginWriteArrayI64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 size = 3;
}

message BeginWriteArrayI64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

message WriteArrayU64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  repeated uint64 array = 3;
  uint32 size = 4;
}

message WriteArrayU64Response {
  int32 status = 1;
}

message BeginWriteArrayU64Request {
  nidevice_grpc.Session session = 1;
  uint32 control = 2;
  uint32 size = 3;
}

message BeginWriteArrayU64Response {
  int32 status = 1;
  ni.data_monikers.Moniker moniker = 2;
}

